### PR TITLE
feat(2464): exclude contact from birthday calendar

### LINKED ISSUE #2464: Exclude Contacts from Birthday-Calendar
**Is your feature request related to a problem? Please describe.**
Sometimes there are contacts that do have birthdays saved, but a user might not care about them in their calendar. (Old Friends, Coworkers, etc)

**Describe the solution you'd like**
I'd like the ability to exclude them from the automatic generation of the birthday-calendar, without having to delete the actual birthday from the contact. There is already a three-dot menu in the ui, it could be added there as a simple button.

### PR DESCRIPTION
**Pull request by sylvainmetayer**
This should close #2464

### CODE REVIEW

File: src/components/ContactDetails.vue [hunk 1]
```diff
@@ -502,6 +509,12 @@ export default {
 		contact() {
 			return this.$store.getters.getContact(this.contactKey)
 		},
+
+		excludeFromBirthdayLabel() {
+			return this.localContact.vCard.hasProperty(this.excludeFromBirthdayKey)
+				? 'Include contact to Birthday Calendar'
```
Comment (tcitworld) [inline src/components/ContactDetails.vue:515]: You need to translate the strings with the `t` function:
```suggestion
				? t('contacts', 'Include contact to Birthday Calendar')
```

File: src/components/ContactDetails.vue [hunk 2]
```diff
@@ -338,6 +344,7 @@ export default {
 			pickedAddressbook: null,
 
 			contactDetailsSelector: '.contact-details',
+			excludeFromBirthdayKey: 'x-nc-exclude-from-birthday-calendar',
```
Comment (tcitworld) [inline src/components/ContactDetails.vue:347]: iCalendar property parameters are usually in upper case, so let's be explicit here, even if the vCard serializer might fix it anyway.
```suggestion
			excludeFromBirthdayKey: 'X-NC-EXCLUDE-FROM-BIRTHDAY-CALENDAR',
```

File: src/components/ContactDetails.vue [hunk 3]
```diff
@@ -502,6 +509,12 @@ export default {
 		contact() {
 			return this.$store.getters.getContact(this.contactKey)
 		},
+
+		excludeFromBirthdayLabel() {
+			return this.localContact.vCard.hasProperty(this.excludeFromBirthdayKey)
+				? t('Include contact to Birthday Calendar')
```
Comment (tcitworld) [inline src/components/ContactDetails.vue:515]:
```suggestion
				? t('contacts', 'Include contact to Birthday Calendar')
```
The function requires the app name as first argument.

Comment (sylvainmetayer) [reply src/components/ContactDetails.vue]: Oops, sorry for the typo ! Just fixed it.

**Comment by sylvainmetayer** [comment]
Is there any tests I can do on this one for the frontend ? 

I'm trying to run tests locally for the server side to validate what I've pushed is valid, but for the contacts side, I haven't find much resources and don't know how I could test my code.

**Comment by sylvainmetayer** [comment]
Hi, @miaulalala or @tcitworld, could one of you please point me to some directions on how to tests this PR for the frontend part, if tests are mandatory for this to be merged ?

**Review by st3iny — CHANGES_REQUESTED** [review]
First of all, thanks a lot for your contribution.

The button needs a version check because the server PR will only be working from 26 onwards.

## Example:
```js
if (parseInt(OC.config.version.split('.')[0]) >= 26) {
}
```

And then toggle it via `v-if`.

File: src/components/ContactDetails.vue (hunk 2 above)
Comment (sylvainmetayer) [reply src/components/ContactDetails.vue:351]: I revert the change to set it lowercase, it seems that the ICAL.Component expose the properties as lowercase.

They are still sent to server as uppercase

**Comment by sylvainmetayer** [comment]
I added a version check on the button as requested

File: src/components/ContactDetails.vue [hunk 4]
```diff
@@ -502,6 +513,16 @@ export default {
 		contact() {
 			return this.$store.getters.getContact(this.contactKey)
 		},
+
+		excludeFromBirthdayLabel() {
+			return this.localContact.vCard.hasProperty(this.excludeFromBirthdayKey)
+				? t('contacts', 'Include contact to Birthday Calendar')
```
Comment (st3iny) [inline src/components/ContactDetails.vue:519]:
```suggestion
				? t('contacts', 'Add contact to Birthday Calendar')
```

**Review by st3iny — COMMENTED** [review]
Tested and works. Thanks again for your contribution :)

There is one minor issue with the wording but otherwise we are good to go.

**Review by st3iny — APPROVED** [review]
LGTM!

One last thing I totally forgot in my last review: Please squash all commits into one before the merge. I'm sorry for all the nitpicking. This will be my last request :see_no_evil:

**Comment by sylvainmetayer** [comment]
I've squashed all commits into a single one, the history should be better now :)

**Pull request merged**